lib/lib-table-cluster/src/test/java/org/sagebionetworks/table/cluster/SQLQueryTest.java

```diff
@@ -799,13 +802,45 @@ public void testSelectViewWithEtagAggregate() throws ParseException{
 	@Test
 	public void testAliasGroupByOrderBy() throws ParseException {
 		sql = "select \"foo\" as \"f\", sum(inttype) as \"i` sum\" from syn123 group by \"f\" order by \"i` sum\" DESC";
-		SqlQuery query = new SqlQueryBuilder(sql)
+		SqlQuery query = new SqlQueryBuilder(sql, userId)
 		.tableSchema(tableSchema)
 		.tableType(EntityType.table)
 		.build();
 		assertEquals("SELECT _C111_ AS `f`, SUM(_C888_) AS `i`` sum` FROM T123 GROUP BY `f` ORDER BY `i`` sum` DESC", query.getOutputSQL());
 	}
 
+	@Test
+	public void testCurrentUserFunctionInWhereClause() throws ParseException {
+		sql = "select COUNT(*) from syn123 where inttype = CURRENT_USER()";
+		Map<String, Object> expectedParameters = Collections.singletonMap("b0", userId);
+		SqlQuery query = new SqlQueryBuilder(sql, userId)
+				.tableSchema(tableSchema)
+				.tableType(EntityType.table)
+				.build();
+		assertEquals("SELECT COUNT(*) FROM T123 WHERE _C888_ = :b0", query.getOutputSQL());
+		assertEquals(expectedParameters.entrySet(), query.getParameters().entrySet());
+	}
+
+	@Test
+	public void testCurrentUserFunctionInAggClause() throws ParseException {
+		sql = "select COUNT(CURRENT_USER()) from syn123";
+		SqlQuery query = new SqlQueryBuilder(sql, userId)
+				.tableSchema(tableSchema)
+				.tableType(EntityType.table)
+				.build();
+		assertEquals("SELECT COUNT(1) FROM T123", query.getOutputSQL());
+	}
+
+	@Test
+	public void testCurrentUserFunctionInSelectClause() throws ParseException {
+		sql = "select CURRENT_USER() from syn123";
+		SqlQuery query = new SqlQueryBuilder(sql, userId)
+				.tableSchema(tableSchema)
+				.tableType(EntityType.table)
+				.build();
+		assertEquals("SELECT 1, ROW_ID, ROW_VERSION FROM T123", query.getOutputSQL());
+	}
+

```

Comment:
You might want to add a test for the getSelectColumns with the currentUser call to check the function return type.

Reply:
Tested the FunctionReturnType in CurrentUserFunctionTest.java

Reply:
I will also add a FunctionReturnType check to to SQLTranslatorUtilsTest